common/djangoapps/util/db.py

```diff
@@ -4,44 +4,119 @@
 from functools import wraps
 import random
 
-from django.db import connection, transaction
+from django.db import DEFAULT_DB_ALIAS, DatabaseError, Error, transaction
 
 
 MYSQL_MAX_INT = (2 ** 31) - 1
 
 
-def commit_on_success_with_read_committed(func):  # pylint: disable=invalid-name
+class CommitOnSuccessManager(object):
```

Comment:
My understanding here: this manager does *not* use savepoints - and either commits all DB queries in a possibly-nested transaction -or- none of them. It differs from `transaction.atomic` in that nested `atomic` calls can cause part of the transaction's queries to still be committed.